python/pyspark/ml/feature.py

```diff
@@ -271,35 +332,35 @@ class _LSHParams(HasInputCol, HasOutputCol):
         typeConverter=TypeConverters.toInt,
     )
 
-    def __init__(self, *args):
+    def __init__(self, *args: Any) -> None:
         super(_LSHParams, self).__init__(*args)
         self._setDefault(numHashTables=1)
 
-    def getNumHashTables(self):
+    def getNumHashTables(self) -> int:
         """
         Gets the value of numHashTables or its default value.
         """
         return self.getOrDefault(self.numHashTables)
 
 
-class _LSH(JavaEstimator, _LSHParams, JavaMLReadable, JavaMLWritable):
+class _LSH(Generic[JM], JavaEstimator[JM], _LSHParams, JavaMLReadable, JavaMLWritable):
```

Comment:
With all annotations in `.py` files, generics should be placed at the last position.